src/main/java/mate/academy/springboot/datajpa/controller/CategoryController.java

```diff
@@ -0,0 +1,48 @@
+package mate.academy.springboot.datajpa.controller;
+
+import lombok.AllArgsConstructor;
+import mate.academy.springboot.datajpa.dto.request.CategoryRequestDto;
+import mate.academy.springboot.datajpa.dto.response.CategoryResponseDto;
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.service.CategoryService;
+import mate.academy.springboot.datajpa.service.mapper.CategoryDtoMapper;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/categories")
+@AllArgsConstructor
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final CategoryDtoMapper mapper;
+
+    @PostMapping
+    public CategoryResponseDto create(
+            @RequestParam CategoryRequestDto requestDto) {
+        return mapper.toDto(categoryService.save(mapper.toModel(requestDto)));
```

Comment:
Not necessarily but my advice:
You have 3 objects with 3 method calls in 1 line. It's a bit hard to read, better split it to 2 lines.
Product product =  categoryService.save(mapper.toModel(requestDto));
return ...